src/routes/stake/UnstakeWarning.svelte

```diff
@@ -0,0 +1,414 @@
+
+		const withdrawResult = await $canisters.icpswapPool.authenticatedActor.withdraw({
+			fee: DEFAULT_LEDGER_FEE,
+			token: CANISTER_ID_ICP_LEDGER,
+			amount: amountToWithdrawE8s
+		} as WithdrawArgs);
+
+		if ('ok' in withdrawResult) {
+			const swapAmount = displayUsFormat(bigintE8sToNumber(withdrawResult.ok), 4);
+			toasts.add(Toast.success(`Successful swap, ${swapAmount} ICP received.`));
+		} else {
+			toasts.add(Toast.error('Failed swap. Please try again.'));
+			throw new Error(`${withdrawResult.err}`);
+		}
+	};
+
+	async function fastUnstake(amount: BigNumber) {
+		if (!$canisters || !$user || $isBusy || amount.isNaN() || !fastUnstakeAmount) return;
+		isBusy.set(true);
+		if ($user.nicpBalance().isGreaterThanOrEqualTo(amount) && amount.isGreaterThan(0)) {
+			try {
+				let amountE8s = numberToBigintE8s(amount);
+				// 1. Approve
+				const spender = {
+					owner: Principal.fromText(CANISTER_ID_ICPSWAP_POOL),
+					subaccount: []
+				} as Account;
+
+				const allowanceResult: Allowance =
+					await $canisters.nicpLedger.anonymousActor.icrc2_allowance({
+						account: { owner: $user.principal, subaccount: [] } as Account,
+						spender
+					} as AllowanceArgs);
+				const allowance = allowanceResult['allowance'];
+				if (numberToBigintE8s(amount) > allowance) {
+					await approveInFastUnstake(spender, amountE8s);
+					amountE8s -= DEFAULT_LEDGER_FEE;
+				}
+
+				// 2. Deposit
+				const amountIn = await depositInFastUnstake(amountE8s);
+				if (!amountIn) return;
+
+				// 3. Swap
+				const amountOut = numberToBigintE8s(fastUnstakeAmount.multipliedBy(BigNumber(0.98)));
+				const amountToWithdrawE8s = await swapInFastUnstake(
+					amountIn.toString(),
+					amountOut.toString()
+				);
+				if (amountToWithdrawE8s === undefined) return;
+
+				// 4. Withdraw
+
+				await withdrawInFastUnstake(amountToWithdrawE8s);
+			} catch (error) {
+				console.log('[fastUnstake] error:', error);
+			}
+		} else {
+			toasts.add(Toast.error('Sorry, there are not enough funds in this account.'));
+		}
+		isBusy.set(false);
+	}
+</script>
+
+<dialog
+	id="unstakeWarningDialog"
+	on:close={() => {
+		inUnstakeWarningMenu.set(false);
+	}}
+	in:fade={{ duration: 500 }}
+>
+	<div class="main-container">
+		<h2>Unstake Confirmation</h2>
+		{#if isFastUnstake}
+			<p>You are currently swapping on the market.</p>
+		{:else}
+			<p>You are currently unstaking with the protocol.</p>
+		{/if}
+		<div class="sum-up-container">
+			<p>Convert {displayUsFormat(BigNumber($inputAmount), 8)} nICP</p>
+			{#if isFastUnstake}
+				<p>
+					Receive {fastUnstakeAmount.toNumber() === 0
+						? '-/-'
+						: displayUsFormat(fastUnstakeAmount, 8)} ICP
+				</p>
+			{:else}
+				<p>
+					Receive {displayUsFormat(
+						computeReceiveAmount(false, BigNumber($inputAmount), exchangeRate),
+						8
+					)} ICP
+				</p>
+			{/if}
+			<p>Effective {isFastUnstake ? 'immediately' : 'in 6 months'}</p>
+		</div>
```

Comment:
Looks good. I think this would be a useful place to let users know about the 0.5% fee if they decide to cancel the delayed unstake (probably best to let them know up front)